spec/gradebook_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rspec'
+require './lib/student'
+require './lib/course'
+require './lib/gradebook'
+
+RSpec.describe Gradebook do
+  it 'exists' do
+    gradebook = Gradebook.new("Mr. Nelson")
+    expect(gradebook).to be_a(Gradebook)
+  end
+
+  it 'has attributes instructor and courses' do
+    gradebook = Gradebook.new("Mr. Nelson")
+    expect(gradebook.instructor).to eq("Mr. Nelson")
+    expect(gradebook.courses).to eq([])
+  end
+
+  it 'add courses to @courses array' do
+    gradebook = Gradebook.new("Mr. Nelson")
+    course1 = Course.new("Calculus", 2)
+    gradebook.add_course(course1)
+    expect(gradebook.courses).to eq([course1])
+  end
+
+  it 'create hash with key as Course and value as array of Student objects' do
+    gradebook = Gradebook.new("Mr. Nelson")
+    course1 = Course.new("Calculus", 2)
+    list_students = ({Calculus: "Bob", Science: "Jim"})
+    expect(gradebook.list_all_students).to eq({Calculus: "Bob", Science: "Jim"})
```

Comment:
according to the interaction pattern, the course object should be the key for this hash (instead of the name of the course).
```suggestion
    expect(gradebook.list_all_students).to eq({course1: "Bob", Science: "Jim"})
```